### PR TITLE
Load and play audio files; rework ResourceManager interface

### PR DESCRIPTION
**Pull request by makometr**
We can store our resources in map-container where name of a resource is a key.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.3)
 
 # set(EXECUTABLE_OUTPUT_PATH "Build")
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
```
Comment (Transliteration) [inline CMakeLists.txt:4]: Hehe, noice

File: Source/ResourceManager.cpp [hunk 2]
```diff
@@ -8,6 +8,11 @@ ResourceManager::ResourceManager(){
 		    throw "Font load failed";
         if (!backgroundGame.loadFromFile("Resources/Pictures/background.png"))
             throw "Gamebackground load failed";
+        
+        if (!figureLanded.loadFromFile("Resources/Sounds/fall.wav"))
+            throw "Audio file 'fall.wav' load failed!";
+        if (!figureLanded.loadFromFile("Resources/Sounds/clear.wav"))
+            throw "Audio file 'clear.wav' load failed!";
```
Comment (Transliteration) [inline Source/ResourceManager.cpp]: Why does same variable load 2 files? Is it store them both? I found nothing about it in documentation(

**Review by Transliteration — APPROVED** [review]
Ez commit - ez life
Approve for c++17

File: Source/ResourceManager.cpp (hunk 2 above)
Comment (makometr) [reply Source/ResourceManager.cpp]: yes, it is ochepyatka

File: Source/ResourceManager.hpp [hunk 3]
```diff
@@ -1,15 +1,37 @@
...
 #include <iostream>
+#include <map>
+#include <array>
 #include <SFML/Graphics.hpp>
 #include <SFML/Audio.hpp>
 
+static constexpr size_t TexturesNumber = 2;
+static constexpr size_t SoundsNumber = 2;
+
+static constexpr auto resourceDirectory = "Resources";
```
Comment (Transliteration) [inline Source/ResourceManager.hpp:11]: **static constexpr auto** weird flex, but okay

File: Source/ResourceManager.hpp [hunk 4]
```diff
@@ -13,29 +13,56 @@ static constexpr auto textureDirectory = "Pictures";
...
 
     sf::Texture& getCellTexture();
-    sf::Texture& getBackground(std::string bcg_name);
+    sf::Texture& getBackground();
+
+private:
+    std::string removeExtension(const std::string& str) const;
+
+    template <typename... T>
+    auto concatStrings(const std::string &divider, const T&... args) const {
```
Comment (Transliteration) [inline Source/ResourceManager.hpp:44]: omg, super weird flex

**Review by Transliteration — APPROVED** [review]
c++over9999 madness
Love it :1st_place_medal:

**Review by Denmey — APPROVED** [review]

**Pull request merged**